persistence/src/main/scala/akka/persistence/EventStoreInterop.scala

```diff
@@ -96,10 +91,13 @@ object EventStoreInterop {
                     State.Consuming(consumer.pure[F]).asLeft[Consumer].leftWiden[State].pure[F]
 
                   case JournalProtocol.ReplayedMessage(persisted) =>
-                    event(persisted).map(event => State.Buffering(Vector(event)).asLeft[Consumer])
+                    val events = Vector(event(persisted))
+                    val state1 = State.Buffering(events): State
+                    state1.asLeft[Consumer].pure[F]
 
                   case JournalProtocol.RecoverySuccess(seqNr) =>
-                    State.Finishing(Vector.empty, seqNr).asLeft[Consumer].leftWiden[State].pure[F]
+                    val state1 = State.Finishing(Vector.empty, seqNr): State
+                    state1.asLeft[Consumer].pure[F]
```

Comment:
I like using `widen` in such cases:
```
val state1 = State.Finishing(Vector.empty, seqNr)
state1.asLeft[Consumer].pure[F].widen
```

Reply:
I like shorter lines if possible and totally fine with : State.

Most important that .widen not always enough and one has to use .widen[State]